providers/aws/sns/topics.go

```diff
@@ -0,0 +1,110 @@
+package sns
+
+import (
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/mitchellh/mapstructure"
+	"go.uber.org/zap"
+)
+
+type Topic struct {
+	ID                        uint `gorm:"primarykey"`
+	AccountID                 string
+	Region                    string
+	Owner                     *string
+	TopicArn                  *string
+	Policy                    *string
+	DeliveryPolicy            *string
+	DisplayName               *string
+	SubscriptionsConfirmed    *int
+	SubscriptionsDeleted      *int
+	SubscriptionsPending      *int
+	EffectiveDeliveryPolicy   *string
+	FifoTopic                 *bool
+	ContentBasedDeduplication *bool
+}
+
+func (Topic) TableName() string {
+	return "aws_sns_topics"
+}
+
+func GetTopicAttributes(c *Client, TopicArn *string) map[string]*string {
+	params := &sns.GetTopicAttributesInput{
+		TopicArn: TopicArn,
+	}
+	output, _ := c.svc.GetTopicAttributes(params)
+	return output.Attributes
+}
+
+func getOrDefault(attrs map[string]*string, keyName string) string {
+	if v, found := attrs[keyName]; found {
+		return *v
+	}
+	return "0"
+}
+
+func (c *Client) transformTopics(values []*sns.Topic) []*Topic {
+	var tValues []*Topic
+	for _, value := range values {
+
+		output := GetTopicAttributes(c, value.TopicArn)
+
+		subsConfirmed, _ := strconv.Atoi(getOrDefault(output, "SubscriptionsConfirmed"))
+		subsDeleted, _ := strconv.Atoi(getOrDefault(output, "SubscriptionsDeleted"))
+		subsPending, _ := strconv.Atoi(getOrDefault(output, "SubscriptionsPending"))
+		isFifo, _ := strconv.ParseBool(getOrDefault(output, "FifoTopic"))
+		isContentBasedDeduped, _ := strconv.ParseBool(getOrDefault(output, "ContentBasedDeduplication"))
+
+		tValue := Topic{
+			AccountID:                 c.accountID,
+			Region:                    c.region,
+			TopicArn:                  value.TopicArn,
+			Policy:                    output["Policy"],
+			DeliveryPolicy:            output["DeliveryPolicy"],
+			DisplayName:               output["DisplayName"],
+			Owner:                     output["Owner"],
+			EffectiveDeliveryPolicy:   output["EffectiveDeliveryPolicy"],
+			SubscriptionsConfirmed:    &subsConfirmed,
+			SubscriptionsDeleted:      &subsDeleted,
+			SubscriptionsPending:      &subsPending,
+			FifoTopic:                 &isFifo,
+			ContentBasedDeduplication: &isContentBasedDeduped,
+		}
+		tValues = append(tValues, &tValue)
+	}
+	return tValues
+}
+
+type TopicConfig struct {
+	Filter string
+}
+
+var TopicTables = []interface{}{
+	&Topic{},
+}
+
+func (c *Client) topics(gConfig interface{}) error {
+	var config sns.ListTopicsInput
+	err := mapstructure.Decode(gConfig, &config)
+	if err != nil {
+		return err
+	}
+	c.db.Where("region", c.region).Where("account_id", c.accountID).Delete(TopicTables...)
```

Comment:
should the delete happen after successfully calling `c.svc.ListTopics(&config)`? If that errors you'd end up with an empty table, right?

Reply:
It's a good questions - I don't think it really matters as the other-way around has it's advantages of not having old data when an error occurs. What I think is important is for this to be consistent across all resources and it's currently not so I'll open an issue and push a fix in other resources. 

From a quick glance I actually think that deleting first is better like implemented here. Because otherwise the delete will be in the [for loop](https://github.com/cloudquery/cloudquery/pull/54/files#diff-76628305ad10b059b8b3dcaac97629b56833706aed226f61538dbb0d76d18c3fR97) which is not the expected behaviour.

Reply:
Good point. Definitely don't want to delete in the loop!

But would be a bummer if you had some transient API error when loading the data from the API and ended up with an empty table

Reply:
Agreed that the behavior is non-ideal; however, you could theoretically just re-run to get the correct results

Reply:
I'm more inclined into the current behaviour but anyway I don't think this change should be part of this PR so I'm good with this one. @James-Quigley Thanks for chiming in! You can open a seperate issue on this one so we can collect a bit more feedback before deciding one way or another.